lib/takeaway.rb

```diff
@@ -0,0 +1,90 @@
+class Takeaway
+
+  CHECKOUT_ERROR = "Total cost entered does not match the sum of your order!"
+
+  attr_reader :menu
+
+  def initialize(menu = Menu.new)
+    @basket = Hash.new(0)
+    @menu = menu
+  end
+
+  def read_menu
+    @menu.dishes.each { |item, cost| print "#{item}: £#{cost}, " }
+  end
+
+  def order(item, quantity=1)
+    if @menu.dishes[item].nil?
+      "#{item} does not exist!"
+    else
+      @basket[item] += quantity
+      "#{quantity}x #{item}(s) added to your basket."
+    end
+  end
+
+  def basket_summary
+    @basket.each do |item, quantity|
+      print "#{item} x#{quantity} = £#{(@menu.dishes[item]*quantity).round(2)}, "
+    end
+  end
+
+  def total_cost
+    @total = []
+    @basket.each do |item, quantity|
+      @total << (@menu.dishes[item]*quantity).round(2)
+    end
+    "Total Cost: £#{(@total.inject(:+)).round(2)}"
+  end
+
+  def checkout(final_cost)
+    if its_correct_amount?(final_cost)
+      send_text
+    else
+      raise CHECKOUT_ERROR
```

Comment:
Style/SignalException: Use `fail` instead of `raise` to signal exceptions. (https://github.com/bbatsov/ruby-style-guide#fail-method)